Posts/Article.0001.ChangeDriveLetter.md

```diff
@@ -0,0 +1,231 @@
+Set-CimInstance: Access is denied.
+```
+
+This error is expected since you are not running PowerShell as an administrator.
+To overcome this error, re-run the command in an elevated session (run as administrator).
+Then your output looks like this:
+
+```powershell-console
+PS C:\Foo> $Drive = Get-CimInstance -ClassName Win32_Volume -Filter "DriveLetter = 'M:'"
+PS C:\Foo> $Drive | Set-CimInstance -Property @{DriveLetter ='X:'}
+PS C:\Foo> Get-Volume | Where-Object FileSystemLabel -eq 'Grateful Dead'
+
+DriveLetter FriendlyName  FileSystemType DriveType HealthStatus OperationalStatus SizeRemaining    Size
+----------- ------------  -------------- --------- ------------ ----------------- -------------    ----
+X           Grateful Dead NTFS           Fixed     Healthy      OK                    591.78 GB 3.64 TB
+```
+
+Changing the drive letter can take a while - so be patient.
+
+And as a final point - you can combine the two property updates in a single call to ``Set-CimInstance``. 
+To revert this drive to the old drive letter (M:\) and it's Label (GD Master) and confirm the change, you can do it like this:
+
+```powershell
+$Drive = Get-CimInstance -ClassName Win32_Volume -Filter "DriveLetter = 'X:'"
+$Drive | Set-CimInstance -Property @{DriveLetter = 'M:'; Label = 'GD Master'}
+```
+
+You can view the resulting change to drive letter and label using ``Get-Volume``.
+The output should look this:
+
+```powershell-console
+PS C:\Foo> Get-Volume | Where-Object FileSystemLabel -match 'GD Master'
+DriveLetter FriendlyName FileSystemType DriveType HealthStatus OperationalStatus SizeRemaining    Size
+----------- ------------ -------------- --------- ------------ ----------------- -------------    ----
+M           GD Master    NTFS           Fixed     Healthy      OK                    591.78 GB 3.64 TB
+```
+
+> **_NOTE:_**  
+One issue you may encounter when you change a drive letter then revert it as shown here.
+It appears that Windows holds on to the old drive latter and does not allow you revert it back immediately.
+Thus you may get a "Set-CimInstance: not available" error message when trying to revert the drive letter.
+To get around this, you have to reboot Windows - it appears just logging off and back on is not adequate.
+
+## Summary
+
+Changing drive letters using PowerShell 7 is simple and straightforward.
+As you can see, using the ``Set-CimInstance`` PowerShell cmdlet to modify writable WMI properties is easy.
+I feel it's more intuitive than making multiple property value assignments (once you you master the hash table). 
+The cool thing is that multiple properties can be modified at one time instead of making multiple value assignments.
+
+## Tip of the Hat
+
+This article was inspired by an earlier Scripting Guys Blog post: <https://devblogs.microsoft.com/scripting/change-drive-letters-and-labels-via-a-simple-powershell-command/>.
+That article was written by the most excellent Ed Wilson - thanks Ed!
```

Comment:
We may want to figure out our linkback style, so this may change.

Reply:
if and when that changes I can update,

Reply:
I made a suggested change to this in the other file. This file needs to be deleted.